packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts

```diff
@@ -0,0 +1,26 @@
+import { SandboxBackendIdResolver } from '../commands/sandbox/sandbox_id_resolver.js';
+import { BackendIdentifierResolver } from './backend_identifier_resolver.js';
+
+/**
+ * Resolves the backend id when branch or stack is passed as an arg, otherwise returns a sandbox backend identifier
+ */
+export class BackendIdentifierResolverWithFallback {
```

Comment:
should this implement `BackendIdentityResolver`?

Reply:
see: https://github.com/aws-amplify/amplify-backend/pull/669#discussion_r1396015448